This file is part of MXE. See LICENSE.md for licensing information.

Contains ad hoc patches for cross building.

From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mark Brand <mabrand@mabrand.nl>
Date: Wed, 29 May 2019 23:01:04 +0200
Subject: [PATCH 1/1] fix mingw build

Author: Chris Matrakidis
taken from https://lists.gnu.org/archive/html/help-glpk/2017-06/msg00039.html

diff --git a/config.h.in b/config.h.in
index 1111111..2222222 100644
--- a/config.h.in
+++ b/config.h.in
@@ -27,4 +27,7 @@
 #undef TLS
 /* thread local storage-class specifier for re-entrancy (if any) */
 
+#undef __WOE__
+/* for windows native compilers */
+
 /* eof */
diff --git a/configure.ac b/configure.ac
index 1111111..2222222 100644
--- a/configure.ac
+++ b/configure.ac
@@ -71,7 +71,13 @@ AC_CHECK_LIB([m], [exp])
 
 dnl Check for <sys/time.h> header
 AC_CHECK_HEADER([sys/time.h],
-   AC_DEFINE([HAVE_SYS_TIME_H], [1], [N/A]))
+   ## Check for 64bit windows timeval type issue (tv_sec is not time_t)
+   AC_COMPILE_IFELSE([AC_LANG_SOURCE([
+      #include <sys/time.h>
+      enum {x = 1/(!!(sizeof(time_t) ==
+                      sizeof(((struct timeval *)0)->tv_sec)))
+      };])], [AC_DEFINE([HAVE_SYS_TIME_H], [1], [N/A])],
+      [[AC_MSG_WARN([sys/time.h not used, tv_sec is not time_t])]]))
 
 dnl Check for gettimeofday function
 AC_CHECK_FUNC([gettimeofday],
@@ -190,6 +196,16 @@ case "${host}" in
 esac
 AC_SUBST([NOUNDEFINED])
 
+AC_MSG_CHECKING([for windows native compiler])
+case "${host}" in *-*-mingw*)
+      AC_MSG_RESULT([yes])
+      AC_DEFINE([__WOE__], [1], [N/A])
+      ;;
+   *)
+      AC_MSG_RESULT([no])
+      ;;
+esac
+
 AC_CONFIG_FILES(
    [src/Makefile examples/Makefile Makefile])
 AC_OUTPUT
